### ISSUE TITLE
Files get mounted as empty directory under linux

### ISSUE THREAD

**Issue by thinking-aloud** [kubernetes/minikube#996]
**BUG REPORT**

**Minikube version**
happened in v0.14.0 and v0.15.0 (others not tested)

**Environment**
- **OS**: Manjaro (Arch Linux x64)
- **VM Driver** : VirtualBox
- **Docker version** : 1.12.5
- **Install tools**: kubectl

**What happened**
Error syncing pod, skipping: failed to "StartContainer" for "gis-data-svc" with RunContainerError: "runContainer: Error response from daemon: rpc error: code = 2 desc = \"oci runtime error: could not synchronise with container process: not a directory\""

**What you expected to happen**
It should work, like it does for minikube under OSX

**How to reproduce it**
I mount a jar into a pod for development. This works under OSX. In linux the error pops up.
I changed my mount point from /app/app.jar to /app/test.jar. This creates an empty directory called test.jar

**additional information**
also happens with kvm as vm-driver

**Comment by korseby** [comment]
I'm getting the same error message when lifting up a ReplicationController that tries to mount a file when it does expect a directory.

`
Error syncing pod, skipping: failed to "StartContainer" for "galaxy-k8s" with RunContainerError: "runContainer: Error response from daemon: rpc error: code = 2 desc = \"oci runtime error: could not synchronise with container process: not a directory\""`

That is because the **mountPath** in **VolumeMounts** and **hostPath** in **volumes** in the yaml-file only accept directories. I would suggest to add **mountFile** and **hostFile** options in order to mount single files into existing directories inside the containers. I suspect that older versions of minikube and/or kubernetes simply ignored that while newer versions are more strict.

Don't know what I am talking about? Just look at the files from @pierrickrogermele here:
https://github.com/pierrickrogermele/vms/tree/master/phn-kube

```
[..]
        volumeMounts:
          - mountPath: "/opt/galaxy_data"               # does work
            name: galaxy-pvc
          - mountPath: "/galaxy/config/galaxy.ini"      # does not work as it points toward a file
            name: galaxy-ini
[..]
      volumes:
        - name: galaxy-pvc                              # does work
          persistentVolumeClaim:
              claimName: galaxy-pvc
        - name: galaxy-ini                              # does not work as it points toward a file
          hostPath:
            path: /phenomenal/github-phnmnl/container-galaxy-k8s-runtime/config/galaxy.ini
[..]
```

There is the `galaxy-k8s-rc-dev.yaml` file that tries to mount single `.ini` and `.xml` into the container. Uncommenting the lines results in the container lifted up without errors. I know that @pierrickrogermele uses macOS while I am on Ubuntu 16.04.

To sum up, I wouldn't classify this as a bug as it is more or less expected behaviour. Ubuntu / Linux seems to be more strict than macOS.

Hey @sneumann and @pcm32, this could be interesting for you as well. :)

**Comment by thinking-aloud** [comment]
What I found was is that the VirtualBox creates a mountpoint called homedir or something that is then mounted. Changing this to the correct name now mounts all the content to the VM. Afterwards it worked.
Strangely in OSX it was mounted into my home dir directly, so the behaviour is somewhat strange.

**Comment by korseby** [comment]
@pcm32 told me that on linux you already have a `/home`-directory on the host, while on macOS it is `/Users` and because of that there are no conflicts. It might help to change `home` to `hosthome` when you are on linux. It would be nice to have some kind of documentation on this. Or at least, I don't know where to find it. ;)

**Comment by thinking-aloud** [comment]
Well, on OSX its /Users/<Username> and on linux /home/<username>.
OSX mounts the mount files from the k8s dir into the home dir. Linux into /homedir. Strangely the linux path of the mount exists in the /home path, but the files all have the size of 4 byte and don't do anything. This is why I first did not find the solution

**Comment by aaron-prindle** [comment]
Closing. We have added the `minikube mount /path/to/dir/to/mount` command for mounting files into minikube.  This allows users to specify the folder to mount and maintains a consistent name.
